course_info/static/course_info/js/directives/BadgeDirective.js

```diff
@@ -0,0 +1,38 @@
+(function () {
+    var app = angular.module('CourseInfo');
+
+    app.directive('badge', function() {
+        return {
+            restrict: 'E',
+            scope: {
+                role: '@',
+            },
+            controller: function($scope) {
+                // convert from role_type_cd to a badge
+                $scope.roleToBadgeMap = {
+                    CLASPART: 'HUID',
+                    COUNTWAY: 'LIBRARY',
+                    EMPLOYEE: 'HUID',
+                    STUDENT: 'HUID',
+                    WIDENER: 'LIBRARY',
+                    XIDHOLDER: 'XID',
+                };
+                $scope.roleToBadge = function(role) {
+                    return $scope.roleToBadgeMap[role] || 'OTHER';
+                };
+
+                // get the css class we want for a badge
+                $scope.badgeToClassMap = {
+                    HUID: 'label-danger',
+                    XID: 'label-primary',
+                    LIBRARY: 'label-success',
+                    OTHER: 'label-warning',
+                }
```

Comment:
AIUI, that was a deliberate choice on @Vittorio2015's part, to not just alias bootstrap classes with our names.